crates/threshold-signature-server/src/validator/tests.rs

```diff
@@ -41,195 +35,81 @@ use entropy_client::{
-            signer.tss_account.0,
-            unsafe_get(&client, hex::encode(NETWORK_PARENT_KEY), port).await,
-        );
+        let key_share = unsafe_get(&client, hex::encode(NETWORK_PARENT_KEY), port).await;
+        assert!(!key_share.is_empty());
     }
 
-    // Get all validators
-    let validators_query = entropy::storage().session().validators();
-    let all_validators = query_chain(&api, &rpc, validators_query, None).await.unwrap().unwrap();
-
-    // Get stash account of a non-signer, to become the new signer
-    // Since we only have 4 nodes in our test setup, this will be the same one the chain chooses
-    let new_signer = all_validators.iter().find(|v| !signer_stash_accounts.contains(v)).unwrap();
-
-    let block_number = TEST_RESHARE_BLOCK_NUMBER;
-    let onchain_reshare_request = OcwMessageReshare {
-        new_signers: vec![new_signer.0.to_vec()],
-        block_number: block_number - 1,
-    };
-
-    run_to_block(&rpc, block_number).await;
-    // Send the OCW message to all TS servers who don't have a chain node
-    let response_results = join_all(
-        validator_ports
-            .iter()
-            .map(|port| {
-                client
-                    .post(format!("http://127.0.0.1:{}/validator/reshare", port))
-                    .body(onchain_reshare_request.clone().encode())
-                    .send()
-            })
-            .collect::<Vec<_>>(),
-    )
-    .await;
-    for response_result in response_results {
-        assert_eq!(response_result.unwrap().text().await.unwrap(), "");
-    }
-
-    for (tss_account, key_share_and_aux_before) in key_shares_before.iter() {
-        let (key_share_before, aux_info_before): KeyShareWithAuxInfo =
-            deserialize(key_share_and_aux_before).unwrap();
-
-        let port = get_port(signers.iter().find(|s| s.tss_account.0 == *tss_account).unwrap());
-        let key_share_and_aux_after =
-            unsafe_get(&client, hex::encode(NETWORK_PARENT_KEY), port).await;
-        let (key_share_after, aux_info_after): KeyShareWithAuxInfo =
-            deserialize(&key_share_and_aux_after).unwrap();
-
-        // Check key share has not yet changed
-        assert_eq!(serialize(&key_share_before).unwrap(), serialize(&key_share_after).unwrap());
-        // Check aux info has not yet changed
-        assert_eq!(serialize(&aux_info_before).unwrap(), serialize(&aux_info_after).unwrap());
-    }
-
-    let new_signers = {
-        let signer_query = entropy::storage().staking_extension().signers();
-        let signer_ids = query_chain(&api, &rpc, signer_query, None).await.unwrap().unwrap();
-        let mut signers = Vec::new();
-        for signer in signer_ids {
-            let query = entropy::storage().staking_extension().threshold_servers(signer);
-            let server_info = query_chain(&api, &rpc, query, None).await.unwrap().unwrap();
-            signers.push(server_info);
+    let mut i = 0;
+    let new_signer_ids = loop {
```

Comment:
```suggestion
    // Wait up to 2min for reshare to complete: check once every second if we have a new set of signers.
    let new_signer_ids = loop {
```

maybe can be helpful for readers?

Reply:
Just out of curiosity, but doesn't `subxt` provide facilities to subscribe to changes to some storage item? Setup a callback that gets called when the event happens?

Reply:
Yep you are right, i think this is the event we need to wait for: https://github.com/entropyxyz/entropy-core/blob/fddd9e4fc41d7526fb361b976a7de96938e97dd1/pallets/staking/src/lib.rs#L374

Reply:
I didn't mean to resolve this conversation - github did it when i pressed 'commit suggestion'

Reply:
It was merely an idea sparked by a remote memory! Feel free to pursue (or not) in a separate PR. :)